### PR TITLE
Clean up and consistently implement the `hsStream` API

### PR DESCRIPTION
**Pull request by dgelessus**
Some `hsStream` methods weren't implemented in all subclasses even though they should have been. This fixes those cases - either implementing the missing methods, or removing them from the base `hsStream` if they actually only make sense for some subclasses.

A few things in the stream classes were implemented twice under different names for no apparent reason, so I removed those duplicates.

Notably, this merges `hsRAMStream` and `hsVectorStream`. Both provided a memory-based read/write stream, but with two different implementations for the dynamic memory buffer. I kept just the `std::vector` version from `hsVectorStream`, because I don't see any advantage to the custom `hsAppender` that `hsRAMStream` used before.

### CODE REVIEW

File: Sources/Plasma/CoreLib/hsStream.cpp [hunk 1]
```diff
@@ -660,88 +624,96 @@ uint32_t  plReadOnlySubStream::GetEOF()
...
+
+void hsRAMStream::FastFwd()
+{
+    fPosition = fVector.size();
 }
 
 void hsRAMStream::Truncate()
 {
-    Reset();
+    fVector.resize(fPosition);
```
Comment (zrax) [inline Sources/Plasma/CoreLib/hsStream.cpp:679]: These two things are not equal...  The old `hsRAMStream::Truncate()` would always reset the stream to 0 size.  Did you check everywhere Truncate was called to see which behavior was expected?

Comment (dgelessus) [reply Sources/Plasma/CoreLib/hsStream.cpp:679]: Oh, good point, I didn't even notice that. Both `hsUNIXStream` and the former `hsVectorStream` implement `Truncate` as "delete all data after current seek position", so I would say the old `hsRAMStream::Truncate` was simply incorrect.

I'll have to check if anything relied on the old `hsRAMStream` behavior...

Comment (dgelessus) [reply Sources/Plasma/CoreLib/hsStream.cpp:679]: Okay, thankfully `hsStream::Truncate` is basically never used. It's only called three times in all of Plasma, and two of those calls are inside other `Truncate` implementations. The one "real" call was on a `hsRAMStream`, *of course*, so I changed it to `Reset`.

File: Sources/Plasma/CoreLib/hsStream.cpp [hunk 2]
```diff
@@ -660,88 +624,96 @@ uint32_t  plReadOnlySubStream::GetEOF()
...
-    return (fBytesRead >= fAppender.Count() * fAppender.ElemSize());
+    return (fPosition >= fVector.size());
 }
 
 uint32_t hsRAMStream::Read(uint32_t byteCount, void * buffer)
 {
-    if (fBytesRead + byteCount > fAppender.Count() * fAppender.ElemSize())
-        byteCount = (fAppender.Count() * fAppender.ElemSize()) - fBytesRead;
+    if (fPosition + byteCount > fVector.size()) {
+//      hsStatusMessageF("Reading past end of hsRAMStream (read %u of %u requested bytes)", fVector.size() - fPosition, byteCount);
```
Comment (Hoikas) [inline Sources/Plasma/CoreLib/hsStream.cpp:635]:
```suggestion
```

**Review by Hoikas — APPROVED** [review]

**Pull request merged**